whatrecord/bin/graph.py

```diff
@@ -0,0 +1,220 @@
+        "--friendly-format",
+        type=str,
+        default="console",
+        help="Output Python object representation instead of JSON",
+    )
+
+    parser.add_argument(
+        '--use-gdb',
+        action="store_true",
+        help="Use metadata derived from the script binary",
+    )
+
+    parser.add_argument(
+        "--expand",
+        action="store_true",
+        help="Expand a substitutions file, as in the msi tool",
+    )
+
+    parser.add_argument(
+        "--highlight",
+        nargs="*",
+        type=str,
+        default=(".*", ),
+        help="Highlight the provided regular expression matches",
+    )
+
+    return parser
+
+
+def _combine_databases(*items: Database) -> Database:
+    for item in items:
+        if not isinstance(item, (Database, LinterResults)):
+            raise ValueError(f"Expected Database, got {type(item)}")
+
+    db = items[0]
+    for item in items[1:]:
+        for record, instance in item.records.items():
+            existing_record = db.records.get(record, None)
+            if not existing_record:
+                db.records[record] = instance
+            else:
+                existing_record.info.update(instance.info)
+                existing_record.metadata.update(instance.metadata)
+                existing_record.fields.update(instance.fields)
+                existing_record.aliases = list(
+                    sorted(set(existing_record.aliases + instance.aliases))
+                )
+                if existing_record.record_type != instance.record_type:
+                    logger.warning(
+                        "Record type mismatch in provided database files: "
+                        "%s %s %s",
+                        record, instance.record_type, existing_record.record_type
+                    )
+
+        db.record_types.update(item.record_types or {})
+
+    return db
+
+
+def main(
+    filenames: List[AnyPath],
+    dbd: Optional[str] = None,
+    standin_directory: Optional[List[str]] = None,
+    macros: Optional[str] = None,
+    friendly: bool = False,
+    use_gdb: bool = False,
+    format: Optional[str] = None,
+    expand: bool = False,
+    friendly_format: str = "console",
+    highlight: Optional[List[str]] = None,
+    only: Optional[List[str]] = None,
+    v3: bool = False,
+):
+    highlight = highlight or []
+    only = only or []
+
+    databases_only = len(filenames) > 1
+    results = [
+        parse_from_cli_args(
+            filename=filename,
+            dbd=dbd,
+            standin_directory=standin_directory,
+            macros=macros,
+            use_gdb=use_gdb,
+            format=format,
+            expand=expand,
+            v3=v3,
+        )
+        for filename in filenames
+    ]
+
+    if databases_only:
+        result = _combine_databases(*results)
+    else:
+        result, = results
+
+    if isinstance(result, (LinterResults, LoadedIoc, Database)):
+        if isinstance(result, LoadedIoc):
```

Comment:
I guess this kind of if/else tree construct is why they've added the match statement in 3.10

Reply:
I'd really like to be able to use it, shame we have to wait for 3.10...